types/react-svg/index.d.ts

```diff
@@ -0,0 +1,34 @@
+// Type definitions for react-svg 2.2
+// Project: https://github.com/atomic-app/react-svg
+// Definitions by: Chen Junda <https://github.com/viccrubs>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import * as React from "react";
+export interface ReactSVGProps {
+  /**
+   * Path to the SVG.
+   */
+  path: string;
+  /**
+   * Function to call after the SVG is injected. Receives the newly injected SVG DOM element as a parameter. Defaults to null.
+   */
+  callback?: (dom: SVGSVGElement) => void;
+  /**
+   * Class name to be added to the SVG. Defaults to ''.
+   */
+  className?: string;
+  /**
+   * Class name to be added to the wrapping div. Defaults to ''.
+   */
+  wrapperClassName?: string;
+  /**
+   * Run any script blocks found in the SVG (always, once, or never). Defaults to never.
+   */
+  evalScripts?: 'always' | 'once' | 'never';
+  /**
+   * Inline styles to be added to the SVG. Defaults to {}.
+   */
+  style?: any;
+}
+export default class ReactSVG extends React.Component<ReactSVGProps> { }
```

Comment:
`require("react-svg")` is itself the class -- it's not an object with a `default` property. See the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default). So this should be:
```ts
declare class ReactSVG extends React.Component<ReactSVGProps> {}
export = ReactSVG;
declare namespace ReactSVG {
    export interface ReactSVGProps { ... }
}
```

Reply:
Thanks but having written like this, `import ... from ...` would not work unless `allowSyntheticDefaultImports` is turned on. To make it work, the class has to be re-exported in the namespace, and it works only with `import { ReactSVG } from 'react-svg'; ` not default export. Is there anyway to achieve both?

current version:

```typescript
import * as React from "react";

declare class ReactSVG extends React.Component<ReactSVG.ReactSVGProps> {}

export = ReactSVG;

declare namespace ReactSVG {

  export class ReactSVG extends React.Component<ReactSVGProps> {}
  export interface ReactSVGProps { ... }
}
```

Reply:
If `import ReactSvg from "react-svg";`works with your module loader, then you should enable `--allowSyntheticDefaultImports` (or `--esModuleInterop`). But for users of other module loaders, it is better to get an error at compile time.
You wouldn't want to "acheive both" because only one import syntax is correct -- they correspond to different values at runtime.
* `import r = require("react-svg")` --> `require("react-svg")`
* `import r from "react-svg"` --> `require("react-svg").default`
* `import { ReactSvg } from "react-svg"` --> `require("react-svg").ReactSvg`

Reply:
OK I've pushed a new commit for it.